portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,32 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+function getRandomQuote() {
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
 
 /**
- * Adds a random greeting to the page.
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addMsgToDOM);
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/** Adds a random quote to the DOM. */
+function addMsgToDOM(msg) {
+  console.log('Adding quote to dom: ' + msg);
```

Comment:
Remember to remove debugging logs.

Reply:
hi kayla, I accidentally created this branch, called "week2_branch".  I meant to put it on the "week2_intro_to_fetch" branch, so I just created another pull request, called "Week 2 intro to fetch", with the changes you requested.